### ISSUE TITLE
Identity security

### ISSUE THREAD

**Issue by GuidoNeele** [aspnet/AspNetCore#10073]
Read this blog post http://www.antradar.com/blog-how-secure-is-asp-net-core-login-page about the security of the identity login page. Think it raises some valid issues. Are you thinking of securing Identity even more out of the box?

**Comment by blowdart** [comment]
This is a difference of opinion between the author and .net.

If an attack is in the position to steal cookies, either via malware or via a proxy with a trusted certificate (because of course you're running on HTTPS) they're in a position to steal the password when it's sent, so a copy of a cookie isn't the "real" problem. This was solvable with token binding, but Google didn't want to implement it and dropped it in Chrome, which basically kills it as a useful feature, especially now that Edge is moving to be based on Chromium. Maybe the edge time will offer up a PR that will integrate back, at which point cookies could be limited to an individual machine, but until that happens there's nothing useful we can do as a mitigation. While the time problem expressed is valid, developers can configure the time for which the cookie contents are valid (the expiry time is baked into the cookie valid, rather than relaying on cookie expiry), but using IP address as a mitigation is frankly one that doesn't work due to proxies and mobile networks, where the external IP changes on a rapid basis. 

User enumeration is different, but an understandable concern. Whilst it's true that you can enumerate usernames through the login screen via the timing oracle they describe you can also enumerate users via the register users screen, so protecting the login screen against it does not offer any real protections.

In summary we are happy with what we have, and don't currently see any real need to change. Of course we reevaluate this on a regular basis.

As an aside the only identity cookie is .AspNetCore.Identity.Application, the others listed do not give you any login capability.

**Comment by GuidoNeele** [comment]
@blowdart Thanks for your quick and clear reply! To be clear I'm happy with what Identity has to offer now, but I bumped into it and had never thought of collecting usernames that way. So was looking for some additional information and discuss it.

**Comment by blowdart** [comment]
It's only quick because I'm in Europe right now, so I suspect our time zones are similar :)
